lib/src/golden_test_adapter.dart

```diff
@@ -301,3 +309,98 @@ class FlutterGoldenTestAdapter extends GoldenTestAdapter {
     return layer.toImage(renderObject.paintBounds);
   }
 }
+
+/// {@template _flutter_golden_test_wrapper}
+/// Similar to [MaterialApp], this widget is used to bootstrap a basic Flutter
+/// application for use in golden tests.
+///
+/// Using [MaterialApp] may introduce unexpected behavior in tests, and can
+/// cause localizations to not be loaded properly. This widget sets up the bare
+/// minimum to get the test to run.
+///
+/// Exposed for internal testing. Do not use this explicitly.
+/// {@endtemplate}
+@protected
+@visibleForTesting
+class FlutterGoldenTestWrapper extends StatelessWidget {
+  /// {@macro _flutter_golden_test_wrapper}
+  const FlutterGoldenTestWrapper({
+    super.key,
+    this.globalConfigTheme,
+    this.variantConfigTheme,
+    this.obscureFont = false,
+    required this.child,
+  });
+
+  /// The theme provided by the global [AlchemistConfig], if any.
+  ///
+  /// See [MaterialApp.theme] for more details.
+  final ThemeData? globalConfigTheme;
+
+  /// The theme provided by the current variant's [GoldensConfig], if any.
+  ///
+  /// See [MaterialApp.theme] for more details.
+  final ThemeData? variantConfigTheme;
+
+  /// Whether the default font family of the resolved theme should be set to an
+  /// obscured font.
+  ///
+  /// See [GoldenTestThemeDataExtensions.applyObscuredFontFamily] for more
+  /// details.
+  final bool obscureFont;
+
+  /// The root widget to wrap.
+  ///
+  /// See [MaterialApp.home] for more details.
+  final Widget child;
+
+  /// Resolves the appropriate theme to use for the current test.
+  ///
+  /// If [obscureFont] is true, the default font family of the resolved theme
+  /// will be set to an obscured font. (See
+  /// [GoldenTestThemeDataExtensions.applyObscuredFontFamily] for more details.)
+  ///
+  /// The returned theme will have its text packages stripped. (See
+  /// [GoldenTestThemeDataExtensions.stripTextPackages] for more details.)
+  ///
+  /// The algorithm is as follows:
+  /// - If a [variantConfigTheme] is provided (through a [GoldensConfig]), use
+  ///   it.
+  /// - Otherwise, if a theme is provided through an [InheritedTheme] (such as
+  ///   through an ancestor [MaterialApp] or [Theme] widget), use it.
+  /// - Otherwise, if a [globalConfigTheme] is provided through an
+  ///   [AlchemistConfig], use it.
+  /// - Otherwise, use the [ThemeData.fallback].
+  ThemeData _resolveThemeOf(BuildContext context) {
+    final hasInheritedTheme =
+        context.findAncestorWidgetOfExactType<Theme>() != null;
+    final inheritedTheme = !hasInheritedTheme ? null : Theme.of(context);
+    final fallbackTheme = ThemeData.fallback();
+
+    var resolvedTheme = variantConfigTheme ??
```

Comment:
Nit: would be nice to have this as final or late final. Although perhaps the workaround sacrifices readability 🤔